### DISCUSSION TITLE
Correlation between Trace, Log and Metrics in Opentelemetry using .Net

### DISCUSSION THREAD

**Discussion by neerajCL** [open-telemetry/opentelemetry-dotnet#4846] [Q&A]
Is there any way to correlate between Trace, Log and Metrics in Opentelemetry using .Net/C#
I am able to correlate between Trace and log using OpenTelemetry in .net and able to send the data to Elastic APM Server.
Same trace id present in Trace and Logs. Is there any way to Correlate between Trace, Log and Metrics in OpenTelemetry using .Net

**Comment by cijothomas** [comment]
Exemplars are a way to correlate metrics to Traces (and to logs). Its an experimental feature today, available in latest pre-release versions only.
https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/docs/metrics/exemplars
https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/docs/metrics/customizing-the-sdk/README.md#exemplars

**Reply by neerajCL** [reply]
Thanks @cijothomas  for your reply. Is the latest pre-release version available for dot-net. If it is available can you share the link.

**Reply by cijothomas** [reply]
https://www.nuget.org/packages/OpenTelemetry/1.6.0-rc.1

**Reply by mrblonde91** [reply]
@cijothomas Any idea what release you're targeting for full release of exemplars? It looks really cool so looking forward to it in prod.

**Reply by cijothomas** [reply]
Unfortunately, we do not have a solid date for that... The spec is still not stable, so that's the main pre-requisite. Of course there are few perf tweaks/tests/fix all todos left as well...

**Reply by cijothomas** [reply]
@mrblonde91 If you are interested, and have the bandwidth, please consider contributing to Exemplars area... We are missing tests completely!, and also would love to see some experiments trying different Filters/ExemplarReservoirs to make the perf better.
(One example I was trying - a Reservoir only storing traceid,spanid and nothing else, primarly for those users who only care about metric->trace correlation!. It can be internal for now along with benchmarks to show perf differences.)
